### PR TITLE
NI-DCPower: Enable fetch and measure LCR, update metadata

### PR DESCRIPTION
**Pull request by olsl21**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nimi-python/blob/master/CONTRIBUTING.md).
- [x] I've updated [CHANGELOG.md](https://github.com/ni/nimi-python/blob/master/CHANGELOG.md) if applicable.
- [x] I've added tests applicable for this pull request

### What does this Pull Request accomplish?

- Create "fancy_fetch_measure_lcr" method template for the public wrapper functions for fetch/measure multiple LCR to add channel info to the returned LCRMeasurement objects
- Rename "fancy_fetch" method template to "fancy_fetch_measure"
- Update NI-DCPower metadata to 22.8.0d88
  - Make the existing "fetch_multiple_lcr" and "measure_multiple_lcr" functions private
    - Create public wrapper functions for them: 'fancy fetch multiple lcr' and 'fancy measure multiple lcr'
    - Use 'fancy_fetch_measure_lcr' method template for these wrapper functions
  - Update docs
- Enable the codegen of fetch/measure multiple LCR functions
- Add system tests for these functions

### What testing has been done?

Local build was successful. All the newly added system tests passed.

### CODE REVIEW

File: generated/nidcpower/nidcpower/session.py [hunk 1]
```diff
@@ -4851,6 +4852,100 @@ def fetch_multiple(self, count, timeout=hightime.timedelta(seconds=1.0)):
...
+            # TODO(olsl21): Retrieving the list of channels in the session on every function call is
+            #  silly because they never change #1776
+            all_channels_in_session = self._get_channel_names(range(self.channel_count))
+
+        channel_names = _converters.expand_channel_string(
+            self._repeated_capability,
+            all_channels_in_session
+        )
+        assert len(channel_names) == 1, "fetch_multiple_lcr only supports one channel at a time"
+        for lcr_measurement_object, in zip(lcr_measurements):
```
[the remark this answers is not in the document]
Comment (sbethur) [reply generated/nidcpower/nidcpower/session.py:4945]: I see the reason now - You're using the same mako template for this method and for `measure_multiple_lcr`.

File: src/nidcpower/metadata/functions.py [hunk 2]
```diff
@@ -1128,6 +1128,146 @@
...
+                    'description': '\nSpecifies the number of measurements to fetch.\n'
+                },
+                'name': 'count',
+                'type': 'ViInt32'
+            },
+            {
+                'direction': 'out',
+                'documentation': {
+                    'description': '\nA list of LCRMeasurement instances.\n',
+                    'table_body': [
```
Comment (sbethur) [inline src/nidcpower/metadata/functions.py:1185]: This table needs to be sourced here due to #1784.

**Review by sbethur — APPROVED** [review]

File: generated/nidcpower/nidcpower/session.py [hunk 3]
```diff
@@ -4851,6 +4852,100 @@ def fetch_multiple(self, count, timeout=hightime.timedelta(seconds=1.0)):
...
+
+                Note:
+                When setting the timeout interval, ensure you take into account any triggers so that the timeout interval is long enough for your application.
+
+
+        Returns:
+            measurements (list of LCRMeasurement): A list of LCRMeasurement instances.
+
+                +-----------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+                | channel               | The channel name associated with this LCR measurement.                                                                                                                                                |
```
Comment (sbethur) [inline generated/nidcpower/nidcpower/session.py:4895]: data type information is missing for these class members.

File: src/nidcpower/templates/session.py/fancy_fetch_lcr.py.mako [hunk 4]
```diff
@@ -0,0 +1,52 @@
+<%page args="f, config, method_template"/>\
+<%
+    '''Forwards to _fetch_multiple_lcr() and _measure_multiple_lcr(), then populate the `channel`
```
[the remark this answers is not in the document]
Comment (olsl21) [reply src/nidcpower/templates/session.py/fancy_fetch_lcr.py.mako:3]: @sbethur Based on my understanding, this filename is referenced by the functions metadata (e.g. 'FancyFetchMultipleLCR' -> 'method_templates' -> 'session_filename': 'fancy_fetch_lcr'), so changing the filename would require changing the functions metadata (from the upstream).

Also, 'fancy_fetch_lcr' is actually named in a similar way to the existing non-LCR method template 'fancy_fetch.py.mako', which is also shared by 'fetch_multiple' and 'measure_multiple', so I would prefer to keep the filename as is, or change the non-LCR method template too.

**Review by marcoskirsch — APPROVED** [review]
awaiting official metadata export

File: src/nidcpower/templates/session.py/fancy_fetch_lcr.py.mako (hunk 4 above)
[the remark this answers is not in the document]
Comment (sbethur) [reply src/nidcpower/templates/session.py/fancy_fetch_lcr.py.mako:3]: > Also, 'fancy_fetch_lcr' is actually named in a similar way to the existing non-LCR method template 'fancy_fetch.py.mako', which is also shared by 'fetch_multiple' and 'measure_multiple


Ah, we definitely want to keep them consistent. The ROI is low to retrofit that. Let's leave it as is.

File: generated/nidcpower/nidcpower/session.py (hunk 3 above)
Comment (olsl21) [reply generated/nidcpower/nidcpower/session.py]: I have updated the functions metadata to add a "type" column to the table in my latest commit.

**Comment by olsl21** [comment]
@marcoskirsch @sbethur I've updated the NI-DCPower metadata to 22.8.0d88

**Pull request merged**